cpp/src/io/parquet/reader_impl_helpers.cpp

```diff
@@ -1154,6 +1184,84 @@ aggregate_reader_metadata::select_columns(
       return path_is_valid;
     };
 
+  // Compares two schema elements to be equal except their number of children
+  auto const compare_schema_elems_except_num_children = [](SchemaElement const& lhs,
```

Comment:
STL has something similar, https://en.cppreference.com/w/cpp/utility/functional/equal_to.
so maybe this can be `equal_to_except_children`
dropped num because we don't check anything about the children.

Reply:
Nice suggestion, `compare_schema....` wasn't my favorite name either.